internal/core/commands/sync_dd_search_data.go

```diff
@@ -75,6 +75,13 @@ func (ch SyncSearchDataCommandHandler) Handle(ctx context.Context, query mediato
 			}
 		}
 
+		def := *definition.R
+		dType := def.DeviceType
+		var deviceType string
+		if dType == nil {
```

Comment:
It looks like `device_definitions.device_type_id` is still nullable in the database. I wish it weren't, but given this I wouldn't want to tear down the app if it's missing.

Reply:
ah good point, making a PR to update `device_type_id` to be not null.